src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -295,14 +299,26 @@ public static X509CRL generateCRL(X500Principal issuer,
                                       PrivateKey issuerPrivateKey,
                                       PublicKey issuerPublicKey,
                                       Date thisUpdate,
-                                      Date nextUpdate)
+                                      Date nextUpdate,
+                                      BigInteger crlNumber,
+                                      List<Map<String, Object>> extensions)
         throws CRLException, IOException, OperatorCreationException, NoSuchAlgorithmException
     {
         X509v2CRLBuilder builder = new JcaX509v2CRLBuilder(issuer, thisUpdate);
         builder.setNextUpdate(nextUpdate);
         builder.addExtension(Extension.authorityKeyIdentifier, false,
                              new JcaX509ExtensionUtils().createAuthorityKeyIdentifier(issuerPublicKey));
-        builder.addExtension(Extension.cRLNumber, false, new CRLNumber(BigInteger.ZERO));
+        builder.addExtension(Extension.cRLNumber, false, new CRLNumber(crlNumber));
+        Extensions bcExtensions = ExtensionsUtils.getExtensionsObjFromMap(extensions);
+        if (bcExtensions != null) {
```

Comment:
If we had a second arity, we could just do this block in that version, and leave it and the conversion above out entirely for the version that didn't take an extension list.

Reply:
sure, I can do that. I hadn't wanted a separate arity for each of the params I've been adding, but I think it does make sense for the extensions one to be separate.

Reply:
oh right, I had this thought before.. The tradeoff is a bunch of repetition, unless I pull out the construction of the CRLBuilder? which I sorta didn't feel was worth it, since these are just for testing? I started working on it, it's not _hard_, but sorta feel like it's more important to wrap up my actual testing for the other PR 😬

Reply:
okay.